xarray/static/css/style.css

```diff
@@ -28,7 +28,7 @@ body.vscode-dark {
 
 .xr-wrap {
   display: block !important;
-  min-width: 300px;
+  min-width: 700px;
```

Comment:
Updating the `min-width` to be `700px` seems to solve this bug, however, now it is the same as the `max-width`. I am not sure if this is okay but thought I would point it out.

Reply:
Awesome! Are there other similar values to compare to in that file? Should we just have `min-width`=700px and max-width even bigger, or is the bug from them being different values somehow?

Reply:
I don't really know what I'm looking at with the css here... Pinging @benbovy and  @jsignell as the original authors of the HTML repr 😁 Can either of you tell us how you would handle this?

Reply:
That width controls what the repr looks like when the area that it has available is small. It basically says the width of the repr cannot be less than 700px. This mostly comes up in jupyterlab when you have a split view. Here's the difference:

main:

![image](https://github.com/pydata/xarray/assets/4806877/be9283b8-b100-499d-a9ff-4315e7460106)

this PR:

![Screenshot from 2024-04-12 09-17-25](https://github.com/pydata/xarray/assets/4806877/66ad3a63-4e43-45e3-8d88-a51be79854b6)

see how the repr goes off screen? There is a horizontal scroll, so you can still get to the far right of the repr, but it's just a little less beautiful.

It seems like this change is related to https://github.com/xarray-contrib/datatree/issues/91. Is there a reproducible example that I can mess around with?

Reply:
I found create_test_datatree so I have a reproducer now.

Reply:
Alright I think I have something:

```diff
diff --git a/xarray/core/formatting_html.py b/xarray/core/formatting_html.py
index bef0ab03..d2181f31 100644
--- a/xarray/core/formatting_html.py
+++ b/xarray/core/formatting_html.py
@@ -364,7 +364,7 @@ def summarize_datatree_children(children: Mapping[str, Any]) -> str:
 
     return "".join(
         [
-            "<div style='display: inline-grid; grid-template-columns: 100%'>",
+            "<div style='display: inline-grid; grid-template-columns: 100%; grid-column: 1 / -1'>",
             children_html,
             "</div>",
         ]
@@ -438,7 +438,7 @@ def _wrap_datatree_repr(r: str, end: bool = False) -> str:
     height = "100%" if end is False else "1.2em"
     return "".join(
         [
-            "<div style='display: inline-grid;'>",
+            "<div style='display: inline-grid; grid-template-columns: 0px 20px auto; width: 100%;'>",
             "<div style='",
             "grid-column-start: 1;",
             "border-right: 0.2em solid;",
@@ -459,9 +459,8 @@ def _wrap_datatree_repr(r: str, end: bool = False) -> str:
             "<div style='",
             "grid-column-start: 3;",
             "'>",
-            "<ul class='xr-sections'>",
             r,
-            "</ul>" "</div>",
+            "</div>",
             "</div>",
         ]
     )
diff --git a/xarray/static/css/style.css b/xarray/static/css/style.css
index 0079e7d3..e0a51312 100644
--- a/xarray/static/css/style.css
+++ b/xarray/static/css/style.css
@@ -28,7 +28,7 @@ body.vscode-dark {
 
 .xr-wrap {
   display: block !important;
-  min-width: 700px;
+  min-width: 300px;
   max-width: 700px;
 }
 
```

Reply:
![image](https://github.com/pydata/xarray/assets/4806877/ea26e408-7ef9-4040-8926-0b0693c76477)

Reply:
Amazing! Thank you @jsignell ! @eni-awowale could you include this and maybe stress-test it a little?

Reply:
Thanks for dropping this in @jsignell ! From testing this on a NASA GESDISC dataset with groups I am still seeing the same issue between a `xr.Dataset` and a `datatree.DataTree`. 
![Screenshot 2024-04-12 at 5 32 08 PM](https://github.com/pydata/xarray/assets/51421921/994c838d-6b01-4ed1-b098-cabb1ebbe08a)

Maybe someone else could try testing it on their end?

Reply:
I had a quick play around taking this branch and applying the changes @jsignell included above. [Here's a link to the GESDISC data](http://tropomi.gesdisc.eosdis.nasa.gov/test/data//S5P_TROPOMI_Level2/S5P_L2__CLOUD__HiR.2/2023/238/S5P_OFFL_L2__CLOUD__20230826T111842_20230826T130011_30405_03_020500_20230828T031255.nc) mentioned by @eni-awowale.

To me, this looks good. But maybe I'm missing something?

<img width="808" alt="Screenshot 2024-04-17 at 10 44 19 AM" src="https://github.com/pydata/xarray/assets/7788154/cad367cd-4b20-4dce-9ac2-9a91a2754c45">

Reply:
I chatted with @eni-awowale and have pushed a commit with the changes from @jsignell to this branch.

Reply:
That looks good to me too - so is do we think this is resolved now?

Reply:
...and I broke some tests. Will fix those now.

Reply:
Fixed: [fdf53f](https://github.com/pydata/xarray/pull/8930/commits/fdf53f5d290c1e1e98197ef5046ce07eecb3e87b)

It looks resolved to me, but want to check sure if @eni-awowale is okay with things as she saw different behaviour.

Reply:
I will check this out as soon as I can!

Reply:
Just tried it out and it's fixed! I will update this to not be a draft

Reply:
@TomNicholas - I think we're good to go.